ui/src/modals/datatype.vue

```diff
@@ -0,0 +1,137 @@
+                <datatype :datatype="datatype"></datatype>
+                <div v-if="apps" style="padding:10px;">
+                    <b-alert show variant="info" v-if="apps.length == 0">There are currently no applications that use this datatype.</b-alert>
+                    <p v-else>The following apps require this datatype as an input.</p>
+                    <div v-for="app in apps" :key="app._id" style="display:inline-block; width: 33%;">
+                        <div style="margin-left: 5px; margin-right: 5px; margin-bottom: 10px;">
+                            <app :app="app" descheight="80px"></app>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <!--<b-tabs class="brainlife-tab" v-model="tab_index"></b-tabs>-->
+    </b-container>
+</div>
+</transition>
+</template>
+<script>
+import Vue from 'vue'
+
+import app from '@/components/app'
+import datatype from '@/components/datatype'
+import datatypetag from '@/components/datatypetag'
+
+export default {
+    components: { 
+        app, datatype, datatypetag
+    },
+    data () {
+        return {
+            datatype: null,
+            apps: null,
+
+            alltags: null,
+            config: Vue.config,
+        } 
+    },
+
+    mounted() {
+        this.$root.$on("datatype.view", id=>{
+            // console.log("requested to view", id);
+            this.load(id);
+        });
+        
+        document.addEventListener("keydown", e => {
+            if (e.keyCode == 27) {
+                this.close();
+            }
+        });
+    },
+
+    methods: {
+        load: function(id) {
+            this.$http.get('datatype', {params: {
+                "find": JSON.stringify({
+                    "_id": id
+                })
+            }}).then(res=>{
+                if (!res.body.count) console.error(`No matching datatype found for ${id}`);
+                else {
+                    this.datatype = res.body.datatypes[0];
+                    
+                    this.$http.get('app', {params: {
+                        "find": JSON.stringify({
+                            //look for apps that uses my datatype as input
+                            "inputs.datatype": this.datatype._id,
+                            removed: false,
+                        }),
+                        populate: 'inputs.datatype outputs.datatype contributors',
+                    }}).then(res=>{
+                        if(!res) return; //TODO notify error?
+                        this.apps = res.body.apps;
+                    });
+                }
+            }).catch(console.error);
+        },
+    
+        close: function() {
+            if(!this.datatype) return;
+            this.$router.replace("/datatypes");
+            this.datatype = null;
+        },
+    }
+}
+
+</script>
+<style scoped>
+.datatype-overlay {
```

Comment:
I think we should refactor CSS needed for the modal display (maybe calle it brainlife-modal?) and store it on warehouse.css?